### PR TITLE
[SERVICES-2633] Finalize transaction generation updates for sdk-core v13

### PR DESCRIPTION
**Pull request by mad2sm0key**
## Reasoning
- use SmartContractTransactionsFactory for generating transactions for the following SCs :  Distribution, Escrow, Fees Collector, Locked Asset Token Wrapper, Metabonding, Price Discovery, Token Unstake, Energy Update
  
## Proposed Changes
- update transaction generation methods
- fix unit tests where needed
- add sender parameter where needed + update resolver methods

## How to test
- N/A

### CODE REVIEW

File: src/modules/escrow/services/escrow.transaction.service.ts [hunk 1]
```diff
@@ -1,41 +1,48 @@
...
-                new AddressValue(Address.fromString(senderAddress)),
-                new AddressValue(Address.fromString(receiverAddress)),
-            ])
-            .withChainID(mxConfig.chainID)
-            .withGasLimit(gasConfig.escrow.cancelTransfer)
-            .buildTransaction()
-            .toPlainObject();
+        return await this.mxProxy.getEscrowSmartContractTransaction(
+            new TransactionOptions({
+                sender: senderAddress,
```
Comment (claudiulataretu) [inline src/modules/escrow/services/escrow.transaction.service.ts:37]: `senderAddress` is not the actual sender for this transaction. This transaction is done by an admin user and it requires 2 arguments for `cancelTransfer` endpoint: `senderAddress` and `receiverAddress`. You still need to add the `userAddress` as argument for this method which will be the admin user

File: src/modules/escrow/services/escrow.transaction.service.ts [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
-
-        return contract.methodsExplicit
-            .withdraw([new AddressValue(Address.fromString(senderAddress))])
-            .withChainID(mxConfig.chainID)
-            .withGasLimit(gasConfig.escrow.withdraw)
-            .buildTransaction()
-            .toPlainObject();
+        return await this.mxProxy.getEscrowSmartContractTransaction(
+            new TransactionOptions({
+                sender: senderAddress,
```
Comment (claudiulataretu) [inline src/modules/escrow/services/escrow.transaction.service.ts:21]: This `senderAddress` is not correct. You still need to have a `userAddress` argument for this endpoint and use it as sender. The `senderAddress` refers to the address who sent funds to `userAddress`

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: src/modules/escrow/services/escrow.transaction.service.ts (hunk 1 above)
Comment (mad2sm0key) [reply src/modules/escrow/services/escrow.transaction.service.ts]: done

File: src/modules/escrow/services/escrow.transaction.service.ts (hunk 2 above)
Comment (mad2sm0key) [reply src/modules/escrow/services/escrow.transaction.service.ts]: done

**Review by cfaur09 — APPROVED** [review]

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**